app/gui/src/dashboard/components/dashboard/column/PathColumn.tsx

```diff
@@ -0,0 +1,235 @@
+
+  const { getAssetNodeById } = state
+
+  const { setCategory } = useCategoriesAPI()
+  const setSelectedKeys = useSetSelectedKeys()
+  const setExpandedDirectoryIds = useSetExpandedDirectoryIds()
+
+  // Path navigation exist only for cloud categories.
+  const { getCategoryByDirectoryId } = useCloudCategoryList()
+
+  // Parents path is a string of directory ids separated by slashes.
+  const splittedPath = parentsPath.split('/').filter(isDirectoryId)
+  const rootDirectoryInPath = splittedPath[0]
+
+  const splittedVirtualParentsPath = virtualParentsPath.split('/')
+  // Virtual parents path is a string of directory names separated by slashes.
+  // To match the ids with the names, we need to remove the first element of the splitted path.
+  // As the first element is the root directory, which is not a virtual parent.
+  const virtualParentsIds = splittedPath.slice(1)
+
+  const { rootDirectoryId } = useUser()
+
+  const navigateToDirectory = useEventCallback((targetDirectory: DirectoryId) => {
+    const targetDirectoryIndex = finalPath.findIndex(({ id }) => id === targetDirectory)
+
+    if (targetDirectoryIndex === -1) {
+      return
+    }
+
+    const pathToDirectory = finalPath
+      .slice(0, targetDirectoryIndex + 1)
+      .map(({ id, categoryId }) => ({ id, categoryId }))
+
+    const rootDirectoryInThePath = pathToDirectory.at(0)
+
+    // This should never happen, as we always have the root directory in the path.
+    // If it happens, it means you've skrewed up
+    invariant(rootDirectoryInThePath != null, 'Root directory id is null')
+
+    // If the target directory is null, we assume that this directory is outside of the current tree (in another category)
+    // Which is the default, because path displays in the recent and trash folders.
+    // But sometimes user might delete a directory with it's whole content, and in that case we'll find it in the tree
+    // because parent is always fetched before children.
+    const targetDirectoryNode = getAssetNodeById(targetDirectory)
+
+    if (targetDirectoryNode == null && rootDirectoryInThePath.categoryId != null) {
+      // we reassign the variable only to make ts happy here.
+      const categoryId = rootDirectoryInThePath.categoryId
+
+      setCategory(categoryId)
+      setExpandedDirectoryIds(pathToDirectory.map(({ id }) => id).concat(targetDirectory))
+    }
+
+    setSelectedKeys(new Set([targetDirectory]))
+  })
+
+  const finalPath = (() => {
+    const result: {
+      id: DirectoryId
+      categoryId: AnyCloudCategory['id'] | null
+      label: AnyCloudCategory['label']
+      icon: AnyCloudCategory['icon']
+    }[] = []
+
+    if (rootDirectoryInPath == null) {
+      return result
+    }
+
+    const rootCategory = getCategoryByDirectoryId(rootDirectoryInPath)
+
+    // If the root category is not found it might mean
+    // that user is no longer have access to this root
+    // Usually this could happen if user was removed from the organization
```

Comment:
`user` -> `the user` both here and on the line above. also optional but you could clarify what "this root" is. "this root directory"?

Reply:
Yeah, you right here, updated the comment